devDocs/startupShutdown.md

```diff
@@ -0,0 +1,166 @@
+# NVDA Starting and Exiting outline
+
+## Ways to start NVDA:
+
+1. For an installed copy:
+    1. Ctrl+Alt+N (Desktop shortcut)
+        - test: `startupShutdownNVDA.Starts from desktop shortcut`
+    1. Automatically via Ease of Access on the Windows sign-in screen (at boot or signing out of a previous session)
+    1. Automatically via Ease of Access on User Account Control (UAC) screens
+    1. Automatically by Ease of Access after signing in to Windows
+1. For an installed copy, portable copy, installer:
+    1. An exiting instance of NVDA starting a new process (see shutting down procedures)
+    1. By running the exe (see cli options too).
+    This can be triggered by a user or external process such as an existing NVDA instance
+        - test: `startupShutdownNVDA.Starts`
+1. For source: eg runnvda.bat
+
+## NVDA can be shutdown by:
+
+1. UI within NVDA, with and without an ExitDialog prompt (uses `triggerNVDAExit`):
+    1. NVDA+q
+        - test: `startupShutdownNVDA.Quits from keyboard, Restarts`
+    1. An input gesture to restart
+    1. After changing some settings (eg installed add-ons or UI language), user prompted on dialog exit.
+    1. Via the NVDA menu -> Exit
+        - test: `startupShutdownNVDA.Quits from menu`
+1. A process sending `WM_QUIT`, eg a new NVDA process starting
+1. A handled crash (directly causes a new process to start, terminates unsafely)
+    - test: `startupShutdownNVDA.Restarts on crash, Restarts on braille crash`
+1. An unhandled crash (terminates unsafely)
+    - requires manual testing/confirmation
+1. An external command which kills the process (terminates unsafely) 
+1. Windows shutting down (terminates unsafely) (uses `wx.EVT_END_SESSION`)
+
+## Manual testing
+Instructions for testing startup / shutdown.
+
+### Start from shortcut
+Prerequisites:
+ - NVDA installed
+ - Shortcut enabled during installation
+
+Steps:
+ 1. Press (or emulate) Ctrl+Alt+N, observe NVDA starts up
+
+Variation:
+- At step 1. A version of NVDA is already running. Observe running version exists before installed version starts up.
+
+### Windows Sign-in screen, automatic start
+Prerequisites:
+ - NVDA installed
+ - Enable "Use NVDA during sign-in"
+
+Steps:
+ 1. Start Windows
+ 1. Observe NVDA announces the Windows sign-in screen
+
+### UAC, automatic start
+Prerequisites:
+ - NVDA installed
+
+Steps:
+
+### Windows Successful sign-in, automatic start
+Prerequisites:
+ - NVDA installed
+ - Enable "Start NVDA after I sign in"
+
+Steps:
+ 1. Start Windows
+ 1. Sign in
+ 1. Observe NVDA starts
+
+### Running the *.exe
+
+Steps:
+ 1. Run the NVDA exe from cmd
+ 1. Observe NVDA starts
+
+Variation:
+- installed copy, portable copy, or installer
+- CLI options

```

Comment:
What options?

Reply:
Well any valid combination of the [ones we accept](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#CommandLineOptions) should be tested, this would be quite a lot of work though. 

Are you suggesting I should link the options or describe testing pathways for each combination?

Reply:
> describe testing pathways for each combination?

No, I don't think that would be relevant here. This should ideally be a guide specific to manually testing the startup / shutdown logic.

However, if there are specific options that are relevant, those should be included. If not, just remove this item.

Reply:
```suggestion
```